### ISSUE TITLE
Maximum recommended data size?

### ISSUE THREAD

**Issue by kiran-peddikuppa** [sindresorhus/electron-store#55]
What is the maximum file size we can store in this. Can i store 100MB file

**Comment by sindresorhus** [comment]
That's not a good idea as everything is currently saved into a single JSON file and needs to be parsed for each operation. Would be interesting to explore a way to have file references, so the JSON file only had a reference to the file which could be stored separately. Maybe it could automatically save it as a separate file if the data is a `Buffer`/`TypedArray`.

However, you can already do manual file references yourself by saving the file and then storing the file path in `electron-store`.

Just curious, what kind of data do you intend to store? What format? Buffers, Base64, etc?

**Comment by callen5914** [comment]
I am interested to know also. I will be storing data based on user settings and race track configurations. Should they all be in separate files? Can this be easily done?

**Comment by stuible** [comment]
I personally have begun using this project to cache Base64 Encoded image renders.  I've found it really begins to slow down around the 100mb mark, especially when each entry is very large (30mb each).  I will experiment more as I integrate the project further.

**Comment by stuible** [comment]
Just to follow up, I decided that this was a very bad idea and was not performant at all. When you approach 50mb, performance is noticeably effected.  The easy workaround was to store the large files as separate files using node fs and then store a record of them in electron-store.

**Comment by harryzcy** [comment]
Can electron-store offer options store files separately? It would be nice if implemented

**Comment by Chinafreak** [comment]
> Can electron-store offer options store files separately? It would be nice if implemented

electron-store does support multiple separately files. See at options -> name & cwd:

https://github.com/sindresorhus/electron-store#name
https://github.com/sindresorhus/electron-store#cwd

**Comment by imxuedi** [comment]
> Just to follow up, I decided that this was a very bad idea and was not performant at all. When you approach 50mb, performance is noticeably effected. The easy workaround was to store the large files as separate files using node fs and then store a record of them in electron-store.

Thank you for your try. My plan is to store some about 10mb cache files, obvioisly, my concern is unnecessary. I will tell here if there are any performance problems.

**Comment by sindresorhus** [comment]
No, it is not meant for that. The whole file is read and written on every change, so it gets slow with even 1 MB: https://github.com/sindresorhus/electron-store#can-i-use-it-for-large-amounts-of-data

Save large data as files yourself and store the paths, and use the `name`/`cwd` options if you want more than one store file.
